scripts/check_doc_links.py

```diff
@@ -71,53 +110,52 @@ def main() -> None:  # pylint: disable=too-many-locals
     session.mount("https://", adapter)
     session.mount("http://", adapter)
 
-    for md_file in all_md_files:
-        print(f"Checking {md_file}...", end="")
-        text = read_file(md_file)
-        m = re.findall(URL_REGEX, text)
-        error = False
-        for url in m:
-            # Add the closing parenthesis if it is msising, as the REGEX is too strict sometimes
-            if "(" in url and ")" not in url:
-                url += ")"
-
-            # Check for HTTP urls
-            if not url.startswith("https") and url not in http_skips:
-                http_links.append((md_file, url))
-                error = True
-
-            # Check for broken links: 200 and 403 codes are admitted
-            if url in url_skips + http_skips:
-                continue
-            try:
-                status_code = session.get(url, timeout=5).status_code
-                if status_code not in (200, 403):
-                    broken_links.append((md_file, url, status_code))
-                    error = True
-            except (
-                requests.exceptions.RetryError,
-                requests.exceptions.ConnectionError,
-            ) as e:
-                broken_links.append((md_file, url, e))
-                error = True
-        print("ERROR" if error else "OK")
-
-    broken_links_str = "\n".join(
-        [f"{url[0]}: {url[1]} ({url[2]})" for url in broken_links]
-    )
-    http_links_str = "\n".join([f"{url[0]}: {url[1]}" for url in http_links])
-
-    if broken_links:
-        print(f"Found broken url in the docs:\n{broken_links_str}")
-
-    if http_links:
-        print(
-            f"Found HTTP urls in the docs:\n{http_links_str}\nTry to use HTTPS equivalent urls or add them to 'http_skips' if not possible"
-        )
-
-    if broken_links or http_links:
-        sys.exit(1)
-    sys.exit(0)
+    # Run all file checks in a thread pool
+    with ThreadPoolExecutor(max_workers=MAX_WORKERS) as executor:
+        futures = []
+        for md_file in all_md_files:
+            print(f"Checking {str(md_file)}...")
+            futures.append(
+                executor.submit(check_file, session, md_file, http_skips, url_skips)
+            )
+
+        # Awaiting for results is blocking
+        print("Awaiting for results...")
+        future_results = [future.result() for future in futures]
+
+        # Get errors
+        for i in future_results:
+            if i[1]:
+                http_links[i[0]] = i[1]
+            if i[2]:
+                broken_links[i[0]] = i[2]
```

Comment:
Having `check_file` returning a `tuple` makes this part non-intuitive as someone has to figure out what the indexes are.